src/main/java/life/qbic/openbis/openbisclient/helper/OpenBisClientHelper.java

```diff
@@ -104,12 +105,22 @@ public static ExperimentTypeFetchOptions fetchExperimentTypesCompletely() {
     return experimentTypeFetchOptions;
   }
 
-  public static List<PropertyType> getPropertiesOfEntityType(IEntityType type) {
-  List<PropertyType> res = new ArrayList<>();
-  List<PropertyAssignment> assignments = type.getPropertyAssignments();
-  for (PropertyAssignment propertyAssignment : assignments) {
-    res.add(propertyAssignment.getPropertyType());
+  public static List<PropertyType> getPropertiesOfEntityType(List<IEntityType> foundTypes) {
```

Comment:
Why do you provide a list of entity types here?
Wouldn't it be 
```suggestion
  public static List<PropertyType> getPropertiesOfEntityTypes(List<IEntityType> foundTypes) {
```
otherwise signature and method name don't go together.

Reply:
Because this is where the abstract part of the entity type fetching starts. the method is used by three different functions that return a searched list of types - but the list is only allowed to contain one type, see the rest of the method.
it's not pretty, maybe it should be changed - will be more code though.

Reply:
Just does not seem right. I think you should change the call of this function rather than the signature of it. Otherwise, you modify the public API of the client -> openbis-client-lib major version has to be increased

Reply:
I'll try to fix it